Sources/NIOEmbedded/Embedded.swift

```diff
@@ -15,13 +15,18 @@
 import Dispatch
 import _NIODataStructures
 import NIOCore
+import NIOConcurrencyHelpers
 
-private final class EmbeddedScheduledTask {
+internal let atomicCounter = NIOAtomic.makeAtomic(value: 0)
```

Comment:
Can we explicitly type this as 64-bit, and make it unsigned? This will give us the opportunity to solve #1541 using the same mechanism (though I don't propose doing so as part of this PR).

Reply:
Equally, embedded scheduled task now has an ID and an insertion order.